### PR TITLE
LWDEV-8824 - Update HFT service to settings service V2

### PR DESCRIPTION
**Pull request by njannink**
- Added settings yaml file
- Add checks to settings
- Updated nuget packages

### CODE REVIEW

File: settings.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    QueuePostfix:
+        settings-key: HFT-QueuePostfix
+
+    ChaosKitty:
+        StateOfChaos:
+            settings-key: HFT-ChaosKittyState
+
+    SagasRabbitMqConnStr:
+        settings-key: HFT-SagasRabbitMqConnStr
```
Comment (polonzer) [inline settings.yaml]: We should use common variable for cqrs/sagas. It is "CqrsRabbitConnString" now.

File: settings.yaml [hunk 2]
```diff
@@ -0,0 +1,129 @@
+HighFrequencyTradingService:
```
Comment (polonzer) [inline settings.yaml:1]: I think it's a great time to refactor our settings section. Remove duplicates, obsolete settings; regroup them; use shared variables.

File: settings.yaml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    ChaosKitty:
+        StateOfChaos:
+            settings-key: HFT-ChaosKittyState
+
+    SagasRabbitMqConnStr:
+        settings-key: HFT-SagasRabbitMqConnStr
+        types:
+            - RabbitMq
+
+    MongoSettings:
```
Comment (polonzer) [inline settings.yaml]: This is the same connection string as OrderStateConnString. We should remove this duplcate.

File: settings.yaml (hunk 1 above)
Comment (njannink) [reply settings.yaml]: renamed

File: settings.yaml (hunk 2 above)
Comment (njannink) [reply settings.yaml:1]: I already removed all unused and obsolete settings

File: settings.yaml (hunk 3 above)
Comment (njannink) [reply settings.yaml]: removed

File: settings.yaml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    IpEndpoint:
+        Port: 
+            settings-key: MatchingEngine-Port
+        Host:
+            settings-key: MatchingEngine-Host
+
+AssetsServiceClient:
+    ServiceUrl: 
+        settings-key: AssetsServiceUrl
+    ExpirationPeriod:
```
Comment (polonzer) [inline settings.yaml]: This setting is not used.

File: settings.yaml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+    LimitOrdersFeed:
+        ConnectionString:
+            settings-key: HFT-LimitOrdersFeed
+            types:
+                - RabbitMq
+        ExchangeName:
+            settings-key: HFT-LimitOrdersExchange
+
+    ApiKeysFeed:
```
Comment (polonzer) [inline settings.yaml]: This setting is not used.

File: settings.yaml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        types:
+            - JsonArray
+
+    Dictionaries:
+        CacheExpirationPeriod:
+            settings-key: HFT-DictionaryCacheExpirationPeriod
+
+    Cache:
+        RedisConfiguration:
+            settings-key: HFT-RedisConfiguration
```
Comment (polonzer) [inline settings.yaml]: This value must be the same for OrderBooksCacheProvider job. Consider using shared value.

File: settings.yaml [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+            types:
+                - Redis
+        ApiKeyCacheInstance:
```
Comment (polonzer) [inline settings.yaml]: I would rename it to something more meaningful, like "ApiKeysPrefix". Or even hardcoded it. I don't see right now why devops should have an ability to change this prefix.

File: settings.yaml [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+            settings-key: HFT-ApiKeyCache
+        ApiKeyCacheKeyPattern:
```
Comment (polonzer) [inline settings.yaml]: I think we can hardcode this setting.

File: settings.yaml [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+            settings-key: HFT-ApiKeyCacheKeyPattern
+        WalletCacheKeyPattern:
```
Comment (polonzer) [inline settings.yaml]: I think we can hardcode this setting.

File: settings.yaml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+            settings-key: HFT-WalletCacheKeyPattern
+        FinanceDataCacheInstance:
+            settings-key: HFT-FinanceDataCache
```
Comment (polonzer) [inline settings.yaml]: This value must be the same for OrderBooksCacheProvider job. Consider using shared value.

File: settings.yaml [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+        OrderBooksCacheKeyPattern:
+            settings-key: HFT-OrderBooksCacheKeyPattern
```
Comment (polonzer) [inline settings.yaml]: This value must be the same for OrderBooksCacheProvider job. Consider using shared value.

File: settings.yaml [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+        FeeCacheInstance:
```
Comment (polonzer) [inline settings.yaml]: This setting is not used.

File: settings.yaml [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+            settings-key: HFT-ApiKeyCacheKeyPattern
+        WalletCacheKeyPattern:
+            settings-key: HFT-WalletCacheKeyPattern
+        FinanceDataCacheInstance:
+            settings-key: HFT-FinanceDataCache
+        OrderBooksCacheKeyPattern:
+            settings-key: HFT-OrderBooksCacheKeyPattern
+        FeeCacheInstance: 
+            settings-key: HFT-FeeCache
+        FeeCacheDuration:
```
Comment (polonzer) [inline settings.yaml]: This setting is not used.

File: settings.yaml [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+            settings-key: HFT-FeeCacheDuration
+        FeeCacheLimitOrderPattern:
```
Comment (polonzer) [inline settings.yaml]: This setting is not used.

File: settings.yaml (hunk 4 above)
Comment (njannink) [reply settings.yaml]: removed

File: settings.yaml (hunk 5 above)
Comment (njannink) [reply settings.yaml]: removed

File: settings.yaml [hunk 15]
```diff
@@ -0,0 +1,95 @@
+HighFrequencyTradingService:
+    Db:
+        LogsConnString:
+            settings-key: LogsConnString
```
Comment (NovichikhinAlexey) [inline settings.yaml]: please use custom key for log connection string.
HFT-LogsConnString

**Review by NovichikhinAlexey — DISMISSED** [review]

File: src/Lykke.Service.HFT.Core/Settings/CacheSettings.cs [hunk 16]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace Lykke.Service.HFT.Core.Settings
+{
+    public class CacheSettings
+    {
+        public string RedisConfiguration { get; set; }
+        public TimeSpan CacheExpirationPeriod { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.HFT.Core/Settings/CacheSettings.cs]: I would add some default value and set the property optional

File: src/Lykke.Service.HFT.Core/Settings/DbSettings.cs [hunk 17]
```diff
@@ -0,0 +1,13 @@
+﻿using Lykke.SettingsReader.Attributes;
+
+namespace Lykke.Service.HFT.Core.Settings
+{
+    public class DbSettings
+    {
+        [AzureBlobCheck]
+        public string LogsConnString { get; set; }
+        public string OrderStateConnString { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.HFT.Core/Settings/DbSettings.cs]: Add the check attribute

**Review by polonzer — APPROVED** [review]

**Pull request merged**